### PR TITLE
[poco] add PostgreSQL & MySQL support

### PR DESCRIPTION
**Pull request by SSE4**
closes: #1909 
closes: #1910 

Specify library name and version:  **poco/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/poco/all/conanfile.py [hunk 1]
```diff
@@ -106,10 +106,6 @@ def config_options(self):
     def configure(self):
         if self.options.enable_apacheconnector:
             raise ConanInvalidConfiguration("Apache connector not supported: https://github.com/pocoproject/poco/issues/1764")
-        if self.options.enable_data_mysql:
```
Comment (uilianries) [inline recipes/poco/all/conanfile.py:109]: Why not removing those options for Poco < 1.10? They affect the package for nothing

Comment (SSE4) [reply recipes/poco/all/conanfile.py:109]: the options are already there, before PR. I am only adding processing of them.

Comment (SSE4) [reply recipes/poco/all/conanfile.py:109]: see https://github.com/conan-io/conan-center-index/blob/80c73584466586afbb85458a5770cf5d6be3e5ca/recipes/poco/all/conanfile.py#L103

**Comment by SSE4** [comment]
seems like Poco just discards transitive dependencies of **libmysqlclient.a**, e.g. OpenSSL, and fails with many undefined references.

**Comment by uilianries** [comment]
@SSE4 library link order?

**Comment by DrewImm** [comment]
If the issue is platform dependent, any chance we could release it for all other platforms and throw a flag switch in for now?

**Comment by uilianries** [comment]
> If the issue is platform dependent, any chance we could release it for all other platforms and throw a flag switch in for now?

@DrewImm Yes, we can throw a ConanInvalidConfiguration exception

**Comment by db4** [comment]
If I take it correctly, the problem is the MacOS build failure. Can we just report an error if `enable_data_mysql` or `enable_data_postgresql` are activated for MacOS? Postgresql is absolutely necessary for any serious use of Poco::Data and Linux & Windows support is much better than nothing.

**Comment by SSE4** [comment]
> If I take it correctly, the problem is the MacOS build failure. Can we just report an error if `enable_data_mysql` or `enable_data_postgresql` are activated for MacOS? Postgresql is absolutely necessary for any serious use of Poco::Data and Linux & Windows support is much better than nothing.

yes, sure, let's do it

**Comment by SSE4** [comment]
after the rebase, I am getting new errors:
```
CMake Error at source_subfolder/Data/MySQL/CMakeLists.txt:15 (add_library):
  Target "DataMySQL" links to target "MySQL::client" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at source_subfolder/Data/PostgreSQL/CMakeLists.txt:15 (add_library):
  Target "DataPostgreSQL" links to target "PostgreSQL::client" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at source_subfolder/Data/MySQL/CMakeLists.txt:15 (add_library):
  Target "DataMySQL" links to target "MySQL::client" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at source_subfolder/Data/PostgreSQL/CMakeLists.txt:15 (add_library):
  Target "DataPostgreSQL" links to target "PostgreSQL::client" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```
need to investigate what caused them

**Comment by SSE4** [comment]
new error:
```
In file included from /Users/jenkins/w/cci_PR-1917@3/.conan/data/libmysqlclient/8.0.17/_/_/package/26f150c431afa7d7dce41b36902531779fe1f841/include/mysql.h:84:
/Users/jenkins/w/cci_PR-1917@3/.conan/data/libmysqlclient/8.0.17/_/_/package/26f150c431afa7d7dce41b36902531779fe1f841/include/mysql/client_plugin.h:108:8: error: definition of type 'MYSQL' conflicts with typedef of the same name
struct MYSQL;
       ^
/Users/jenkins/w/cci_PR-1917@3/.conan/data/poco/1.9.3/_/_/build/d5bb8bea3204abf375ca102326578b0a5940e957/source_subfolder/Data/MySQL/include/Poco/Data/MySQL/MySQLException.h:26:25: note: 'MYSQL' declared here
typedef struct st_mysql MYSQL;
```

**Comment by SSE4** [comment]
seems to be the same issue as https://github.com/pocoproject/poco/issues/2744
poco 1.9.x isn't compatible with MySQL client 8.0 and above
the support has been added for poco 1.10.x: https://github.com/pocoproject/poco/commit/c04a1f28d94544205c99f27868f40c32a4c839f8#diff-441831c6b31530e0b1633c5d8bb20cad193ac664809edb52f8ec7b27bb2ed213
I am disabling MySQL for 1.9.x for now.

**Comment by SSE4** [comment]
finally, passed and ready for the review!

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Pull request merged**